raiden/exceptions.py

```diff
@@ -157,22 +157,6 @@ class InvalidToken(RaidenError):
     """ Raised if the token does not follow the ERC20 standard """
 
 
-class InvalidTokenAddress(RaidenError):
```

Comment:
Instead of removing these, you could make `InvalidTokenAddress` a subclass of `BrokenPreconditionError`. Fine grained exceptions are usually a nice thing, e.g. you could mark wrong input fields in the UI or add error codes to exceptions and pass them to API users.

Reply:
Will do.

Reply:
No, I'm not sure.  When the API raises these exceptions, that should not crash the node.

Also I've seen that the proxy should not raise specific errors https://github.com/raiden-network/raiden/issues/4878#issuecomment-531728260 I'm seeing another contradiction here.

Reply:
@rakanalh, in the end, are proxies allowed to raise specific errors?  If yes, in which circumstances?  I'll document and follow what you say.